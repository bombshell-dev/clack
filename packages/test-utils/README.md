# `@clack/test-utils`

A bunch of useful test utiltiies for use inside clack.
