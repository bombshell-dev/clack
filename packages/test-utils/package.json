{
  "name": "@clack/test-utils",
  "private": true,
  "version": "1.0.0-alpha.1",
  "type": "module",
  "main": "./dist/index.js",
  "module": "./dist/index.js",
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "default": "./dist/index.js"
    },
    "./package.json": "./package.json"
  },
  "types": "./dist/index.d.ts",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/bombshell-dev/clack.git",
    "directory": "packages/test-utils"
  },
  "bugs": {
    "url": "https://github.com/bombshell-dev/clack/issues"
  },
  "homepage": "https://github.com/bombshell-dev/clack/tree/main/packages/test-utils#readme",
  "files": ["dist", "CHANGELOG.md"],
  "author": {
    "name": "James Garbutt",
    "url": "https://github.com/43081j"
  },
  "license": "MIT",
  "packageManager": "pnpm@9.14.2",
  "scripts": {
    "build": "tsc",
    "prepack": "pnpm build"
  },
  "dependencies": {
  },
  "devDependencies": {
  }
}
