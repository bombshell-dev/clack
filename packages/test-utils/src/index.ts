export { MockReadable } from './mock-readable.js';
export { MockWritable } from './mock-writable.js';

export function nextTick(): Promise<void> {
	return new Promise<void>((resolve) => {
		setTimeout(resolve, 0);
	});
}
