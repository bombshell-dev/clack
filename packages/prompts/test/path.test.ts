import { vol } from 'memfs';
import { afterAll, afterEach, beforeAll, beforeEach, describe, expect, test, vi } from 'vitest';
import * as prompts from '../src/index.js';
import { MockReadable, MockWritable } from './test-utils.js';

vi.mock('node:fs');

describe.each(['true', 'false'])('text (isCI = %s)', (isCI) => {
	let originalCI: string | undefined;
	let output: MockWritable;
	let input: MockReadable;

	beforeAll(() => {
		originalCI = process.env.CI;
		process.env.CI = isCI;
	});

	afterAll(() => {
		process.env.CI = originalCI;
	});

	beforeEach(() => {
		output = new MockWritable();
		input = new MockReadable();
		vol.reset();
		vol.fromJSON(
			{
				'./foo/bar.txt': '1',
				'./foo/baz.text': '2',
				'./hello/world.jpg': '3',
				'./hello/john.jpg': '4',
				'./hello/jeanne.png': '5',
				'./root.zip': '6',
				'./bar': '7',
			},
			'/tmp'
		);
	});

	afterEach(() => {
		vi.restoreAllMocks();
	});

	test('renders message', async () => {
		const result = prompts.path({
			message: 'foo',
			input,
			output,
			root: '/tmp/',
		});

		input.emit('keypress', '', { name: 'return' });

		const value = await result;

		expect(output.buffer).toMatchSnapshot();
		expect(value).toBe('/tmp/bar');
	});

	test('can cancel', async () => {
		const result = prompts.path({
			message: 'foo',
			root: '/tmp/',
			input,
			output,
		});

		input.emit('keypress', 'escape', { name: 'escape' });

		const value = await result;

		expect(prompts.isCancel(value)).toBe(true);
		expect(output.buffer).toMatchSnapshot();
	});

	test('renders cancelled value if one set', async () => {
		const result = prompts.path({
			message: 'foo',
			input,
			output,
			root: '/tmp/',
		});

		input.emit('keypress', 'x', { name: 'x' });
		input.emit('keypress', 'y', { name: 'y' });
		input.emit('keypress', '', { name: 'escape' });

		const value = await result;

		expect(prompts.isCancel(value)).toBe(true);
		expect(output.buffer).toMatchSnapshot();
	});

	test('renders submitted value', async () => {
		const result = prompts.path({
			message: 'foo',
			root: '/tmp/',
			input,
			output,
		});

		input.emit('keypress', 'b', { name: 'b' });
		input.emit('keypress', 'a', { name: 'a' });
		input.emit('keypress', '', { name: 'return' });

		const value = await result;

		expect(value).toBe('/tmp/bar');
		expect(output.buffer).toMatchSnapshot();
	});

	test('cannot submit unknown value', async () => {
		const result = prompts.path({
			message: 'foo',
			root: '/tmp/',
			input,
			output,
		});

		input.emit('keypress', '_', { name: '_' });
		input.emit('keypress', '', { name: 'return' });
		input.emit('keypress', '', { name: 'h', ctrl: true });
		input.emit('keypress', 'b', { name: 'b' });
		input.emit('keypress', '', { name: 'return' });

		const value = await result;

		expect(value).toBe('/tmp/bar');
		expect(output.buffer).toMatchSnapshot();
	});

	test('initialValue sets the value', async () => {
		const result = prompts.path({
			message: 'foo',
			initialValue: '/tmp/bar',
			root: '/tmp/',
			input,
			output,
		});

		input.emit('keypress', '', { name: 'return' });

		const value = await result;

		expect(value).toBe('/tmp/bar');
		expect(output.buffer).toMatchSnapshot();
	});

	test('validation errors render and clear', async () => {
		const result = prompts.path({
			message: 'foo',
			root: '/tmp/',
			validate: (val) => (val !== '/tmp/bar' ? 'should be /tmp/bar' : undefined),
			input,
			output,
		});

		// to match `root.zip`
		input.emit('keypress', 'r', { name: 'r' });
		input.emit('keypress', '', { name: 'return' });
		// delete what we had
		input.emit('keypress', '', { name: 'h', ctrl: true });
		input.emit('keypress', 'b', { name: 'b' });
		input.emit('keypress', '', { name: 'return' });

		const value = await result;

		expect(value).toBe('/tmp/bar');
		expect(output.buffer).toMatchSnapshot();
	});

	test('validation errors render and clear (using Error)', async () => {
		const result = prompts.path({
			message: 'foo',
			root: '/tmp/',
			validate: (val) => (val !== '/tmp/bar' ? new Error('should be /tmp/bar') : undefined),
			input,
			output,
		});

		// to match `root.zip`
		input.emit('keypress', 'r', { name: 'r' });
		input.emit('keypress', '', { name: 'return' });
		// delete what we had
		input.emit('keypress', '', { name: 'h', ctrl: true });
		input.emit('keypress', 'b', { name: 'b' });
		input.emit('keypress', '', { name: 'return' });

		const value = await result;

		expect(value).toBe('/tmp/bar');
		expect(output.buffer).toMatchSnapshot();
	});
});
