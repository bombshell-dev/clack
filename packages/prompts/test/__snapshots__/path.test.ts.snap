// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`text (isCI = false) > can cancel 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m○[22m [2m/tmp/root.zip[22m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[31m■[39m  foo
[90m│[39m  [9m[2m/tmp/[22m[29m",
  "
",
  "<cursor.show>",
]
`;

exports[`text (isCI = false) > cannot submit unknown value 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m○[22m [2m/tmp/root.zip[22m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=3>",
  "<erase.down>",
  "[36m│[39m  [2mSearch:[22m /tmp/_█
[36m│[39m  [33mNo matches found[39m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[33m▲[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/_█
[36m│[39m  [33mNo matches found[39m
[33m│[39m  [33mPlease select a path[39m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m○[22m [2m/tmp/root.zip[22m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=3>",
  "<erase.down>",
  "[36m│[39m  [2mSearch:[22m /tmp/b█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32m◇[39m  foo
[90m│[39m  [2m/tmp/bar[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`text (isCI = false) > initialValue sets the value 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/bar█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32m◇[39m  foo
[90m│[39m  [2m/tmp/bar[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`text (isCI = false) > renders cancelled value if one set 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m○[22m [2m/tmp/root.zip[22m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=3>",
  "<erase.down>",
  "[36m│[39m  [2mSearch:[22m /tmp/x█
[36m│[39m  [33mNo matches found[39m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=3>",
  "<erase.line><cursor.left count=1>",
  "[36m│[39m  [2mSearch:[22m /tmp/xy█",
  "<cursor.down count=3>",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[31m■[39m  foo
[90m│[39m  [9m[2m/tmp/xy[22m[29m",
  "
",
  "<cursor.show>",
]
`;

exports[`text (isCI = false) > renders message 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m○[22m [2m/tmp/root.zip[22m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32m◇[39m  foo
[90m│[39m  [2m/tmp/bar[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`text (isCI = false) > renders submitted value 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m○[22m [2m/tmp/root.zip[22m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=3>",
  "<erase.down>",
  "[36m│[39m  [2mSearch:[22m /tmp/b█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=3>",
  "<erase.line><cursor.left count=1>",
  "[36m│[39m  [2mSearch:[22m /tmp/ba█",
  "<cursor.down count=3>",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32m◇[39m  foo
[90m│[39m  [2m/tmp/bar[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`text (isCI = false) > validation errors render and clear (using Error) 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m○[22m [2m/tmp/root.zip[22m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=3>",
  "<erase.down>",
  "[36m│[39m  [2mSearch:[22m /tmp/r█
[36m│[39m  [32m●[39m /tmp/root.zip
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[33m▲[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/r█
[33m│[39m  [33mshould be /tmp/bar[39m
[36m│[39m  [32m●[39m /tmp/root.zip
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/█
[36m│[39m  [2m○[22m [2m/tmp/bar[22m
[36m│[39m  [32m●[39m /tmp/root.zip
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=3>",
  "<erase.down>",
  "[36m│[39m  [2mSearch:[22m /tmp/b█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32m◇[39m  foo
[90m│[39m  [2m/tmp/bar[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`text (isCI = false) > validation errors render and clear 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m○[22m [2m/tmp/root.zip[22m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=3>",
  "<erase.down>",
  "[36m│[39m  [2mSearch:[22m /tmp/r█
[36m│[39m  [32m●[39m /tmp/root.zip
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[33m▲[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/r█
[33m│[39m  [33mshould be /tmp/bar[39m
[36m│[39m  [32m●[39m /tmp/root.zip
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/█
[36m│[39m  [2m○[22m [2m/tmp/bar[22m
[36m│[39m  [32m●[39m /tmp/root.zip
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=3>",
  "<erase.down>",
  "[36m│[39m  [2mSearch:[22m /tmp/b█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32m◇[39m  foo
[90m│[39m  [2m/tmp/bar[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`text (isCI = true) > can cancel 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m○[22m [2m/tmp/root.zip[22m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[31m■[39m  foo
[90m│[39m  [9m[2m/tmp/[22m[29m",
  "
",
  "<cursor.show>",
]
`;

exports[`text (isCI = true) > cannot submit unknown value 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m○[22m [2m/tmp/root.zip[22m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=3>",
  "<erase.down>",
  "[36m│[39m  [2mSearch:[22m /tmp/_█
[36m│[39m  [33mNo matches found[39m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[33m▲[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/_█
[36m│[39m  [33mNo matches found[39m
[33m│[39m  [33mPlease select a path[39m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m○[22m [2m/tmp/root.zip[22m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=3>",
  "<erase.down>",
  "[36m│[39m  [2mSearch:[22m /tmp/b█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32m◇[39m  foo
[90m│[39m  [2m/tmp/bar[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`text (isCI = true) > initialValue sets the value 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/bar█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32m◇[39m  foo
[90m│[39m  [2m/tmp/bar[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`text (isCI = true) > renders cancelled value if one set 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m○[22m [2m/tmp/root.zip[22m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=3>",
  "<erase.down>",
  "[36m│[39m  [2mSearch:[22m /tmp/x█
[36m│[39m  [33mNo matches found[39m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=3>",
  "<erase.line><cursor.left count=1>",
  "[36m│[39m  [2mSearch:[22m /tmp/xy█",
  "<cursor.down count=3>",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[31m■[39m  foo
[90m│[39m  [9m[2m/tmp/xy[22m[29m",
  "
",
  "<cursor.show>",
]
`;

exports[`text (isCI = true) > renders message 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m○[22m [2m/tmp/root.zip[22m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32m◇[39m  foo
[90m│[39m  [2m/tmp/bar[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`text (isCI = true) > renders submitted value 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m○[22m [2m/tmp/root.zip[22m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=3>",
  "<erase.down>",
  "[36m│[39m  [2mSearch:[22m /tmp/b█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=3>",
  "<erase.line><cursor.left count=1>",
  "[36m│[39m  [2mSearch:[22m /tmp/ba█",
  "<cursor.down count=3>",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32m◇[39m  foo
[90m│[39m  [2m/tmp/bar[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`text (isCI = true) > validation errors render and clear (using Error) 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m○[22m [2m/tmp/root.zip[22m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=3>",
  "<erase.down>",
  "[36m│[39m  [2mSearch:[22m /tmp/r█
[36m│[39m  [32m●[39m /tmp/root.zip
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[33m▲[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/r█
[33m│[39m  [33mshould be /tmp/bar[39m
[36m│[39m  [32m●[39m /tmp/root.zip
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/█
[36m│[39m  [2m○[22m [2m/tmp/bar[22m
[36m│[39m  [32m●[39m /tmp/root.zip
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=3>",
  "<erase.down>",
  "[36m│[39m  [2mSearch:[22m /tmp/b█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32m◇[39m  foo
[90m│[39m  [2m/tmp/bar[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`text (isCI = true) > validation errors render and clear 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m○[22m [2m/tmp/root.zip[22m
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=3>",
  "<erase.down>",
  "[36m│[39m  [2mSearch:[22m /tmp/r█
[36m│[39m  [32m●[39m /tmp/root.zip
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[33m▲[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/r█
[33m│[39m  [33mshould be /tmp/bar[39m
[36m│[39m  [32m●[39m /tmp/root.zip
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[36m◆[39m  foo
[36m│[39m
[36m│[39m  [2mSearch:[22m /tmp/█
[36m│[39m  [2m○[22m [2m/tmp/bar[22m
[36m│[39m  [32m●[39m /tmp/root.zip
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=7>",
  "<cursor.down count=3>",
  "<erase.down>",
  "[36m│[39m  [2mSearch:[22m /tmp/b█
[36m│[39m  [32m●[39m /tmp/bar
[36m│[39m  [2m[2m↑/↓[22m[2m to select • [2mEnter:[22m[2m confirm • [2mType:[22m[2m to search[22m
[36m└[39m",
  "<cursor.backward count=999><cursor.up count=6>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32m◇[39m  foo
[90m│[39m  [2m/tmp/bar[22m",
  "
",
  "<cursor.show>",
]
`;
